In [1]:
import pandas as pd
import numpy as np

#  Reading raw excel file of schools data
- Skipping for two unecessary row

In [2]:
school_data = pd.read_excel("data/raw-schools_and_Student.xlsx",skiprows=1)

In [3]:
school_data.head()

School_Code District_name VDC/Municipality_name       School_name  \
0          NaN           NaN                   NaN               NaN   
1          NaN           NaN                   NaN               NaN   
2  690430007.0        ACHHAM                Kuskot  Adarsha Ni Ma Vi   
3  690700001.0        ACHHAM             Timilsain     Adarsha Pra V   
4  690700003.0        ACHHAM             Timilsain   Alaknanda Ma Vi   

   Ward No.   Location School running grades and types Unnamed: 7 Unnamed: 8  \
0       NaN        NaN                             NaN        NaN        NaN   
1       NaN        NaN                             ecd     class1     class2   
2       2.0     Kuskot                               0          1          1   
3       4.0  Badchauki                              12          1          1   
4       5.0  Tinilsain                               1          1          1   

  Unnamed: 9     ...      Unnamed: 117 Unnamed: 118 Unnamed: 119 Unnamed: 120  \
0        NaN     ...                 B          NaN            T          NaN   
1     class3     ...              2070       2069.0         2070       2069.0   
2          1     ...                 0          0.0            0          0.0   
3          1     ...                 0          0.0            0          0.0   
4          1     ...                 0          0.0            0          0.0   

  Grand Total Unnamed: 122 Unnamed: 123 Unnamed: 124 Unnamed: 125 Unnamed: 126  
0           G          NaN            B          NaN            T          NaN  
1        2070       2069.0         2070       2069.0         2070       2069.0  
2          83         89.0          104         95.0          187        184.0  
3          46         45.0           36         35.0           82         80.0  
4         255        201.0          205        183.0          460        384.0  

[5 rows x 127 columns]

In [4]:
## columns in dataframe
school_data.columns

Index([                    u'School_Code',                   u'District_name',
                 u'VDC/Municipality_name',                     u'School_name',
                              u'Ward No.',                        u'Location',
       u'School running grades and types',                      u'Unnamed: 7',
                            u'Unnamed: 8',                      u'Unnamed: 9',
       ...
                          u'Unnamed: 117',                    u'Unnamed: 118',
                          u'Unnamed: 119',                    u'Unnamed: 120',
                           u'Grand Total',                    u'Unnamed: 122',
                          u'Unnamed: 123',                    u'Unnamed: 124',
                          u'Unnamed: 125',                    u'Unnamed: 126'],
      dtype='object', length=127)

In [5]:
type(school_data['School_Code'][0])

numpy.float64

In [6]:
school_data.describe()

/Users/BIKESHKAWAN/anaconda/envs/python2env/lib/python2.7/site-packages/numpy/lib/function_base.py:3823: RuntimeWarning: Invalid value encountered in percentile
  RuntimeWarning)


School_Code      Ward No.   Unnamed: 20   Unnamed: 22   Unnamed: 24  \
count  3.623700e+04  36129.000000  36238.000000  36238.000000  36238.000000   
mean   3.592910e+08      5.462316     17.919725     17.845025     35.707655   
std    2.104433e+08      3.779288     20.478407     20.694266     35.297803   
min    1.302000e+04      0.000000      0.000000      0.000000      0.000000   
25%             NaN           NaN           NaN           NaN           NaN   
50%             NaN           NaN           NaN           NaN           NaN   
75%             NaN           NaN           NaN           NaN           NaN   
max    7.504100e+08     99.000000   2069.000000   2069.000000   2069.000000   

        Unnamed: 26   Unnamed: 28   Unnamed: 30   Unnamed: 32   Unnamed: 34  \
count  36238.000000  36238.000000  36238.000000  36238.000000  36238.000000   
mean      13.279210     13.005353     26.227468     12.356891     12.068078   
std       16.774551     16.815754     26.704605     16.226990     16.855660   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%             NaN           NaN           NaN           NaN           NaN   
50%             NaN           NaN           NaN           NaN           NaN   
75%             NaN           NaN           NaN           NaN           NaN   
max     2069.000000   2069.000000   2069.000000   2069.000000   2069.000000   

           ...       Unnamed: 108  Unnamed: 110  Unnamed: 112  Unnamed: 114  \
count      ...       36238.000000   2305.000000  36238.000000  36238.000000   
mean       ...           5.977123     42.045987      2.793145      5.410481   
std        ...          32.031906     57.406518     19.063826     30.600811   
min        ...           0.000000      1.000000      0.000000      0.000000   
25%        ...                NaN           NaN           NaN           NaN   
50%        ...                NaN           NaN           NaN           NaN   
75%        ...                NaN           NaN           NaN           NaN   
max        ...        2069.000000   2069.000000   2069.000000   2069.000000   

       Unnamed: 116  Unnamed: 118  Unnamed: 120  Unnamed: 122  Unnamed: 124  \
count  36238.000000  36238.000000  36238.000000  36238.000000  36238.000000   
mean       5.638473      5.749131     11.330509    109.646642    107.826315   
std       29.940377     32.937230     58.474786    128.009425    132.302272   
min        0.000000      0.000000      0.000000      0.000000      0.000000   
25%             NaN           NaN           NaN           NaN           NaN   
50%             NaN           NaN           NaN           NaN           NaN   
75%             NaN           NaN           NaN           NaN           NaN   
max     2069.000000   2069.000000   2069.000000   2252.000000   3198.000000   

       Unnamed: 126  
count  36238.000000  
mean     217.415862  
std      254.702387  
min        0.000000  
25%             NaN  
50%             NaN  
75%             NaN  
max     5410.000000  

[8 rows x 56 columns]

In [7]:
school_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 36239 entries, 0 to 36238
Columns: 127 entries, School_Code to Unnamed: 126
dtypes: float64(56), object(71)
memory usage: 35.1+ MB


In [8]:
# school_data = pd.DataFrame?

# Making copy of original DataFrame with necessary columns 

In [9]:
school_data_df_copy = pd.DataFrame(data=school_data,columns=['School_name','District_name','Location','VDC/Municipality_name','Ward No.','Unnamed: 125'])

#### We can see first two rows are  NaN Rows and we need to drop them

In [10]:
school_data_df_copy.head()

School_name District_name   Location VDC/Municipality_name  Ward No.  \
0               NaN           NaN        NaN                   NaN       NaN   
1               NaN           NaN        NaN                   NaN       NaN   
2  Adarsha Ni Ma Vi        ACHHAM     Kuskot                Kuskot       2.0   
3     Adarsha Pra V        ACHHAM  Badchauki             Timilsain       4.0   
4   Alaknanda Ma Vi        ACHHAM  Tinilsain             Timilsain       5.0   

  Unnamed: 125  
0            T  
1         2070  
2          187  
3           82  
4          460

In [11]:
school_data_df_copy.tail()

School_name District_name      Location  \
36234              U Ma V Sundarpur      UDAYAPUR     Sundarpur   
36235         Udaya Pra V Salthumki      UDAYAPUR        Khanbu   
36236  Udayapur Little Heaven E B S      UDAYAPUR    Galphariya   
36237         Unik Ebs Shishu Sahan      UDAYAPUR  Gurungchauri   
36238       Yuba Ni Ma V Dhukurthum      UDAYAPUR        Anptar   

      VDC/Municipality_name  Ward No. Unnamed: 125  
36234             Sundarpur       2.0          723  
36235                Khanbu       1.0           62  
36236                Khanbu       1.0           74  
36237              Thoksila       7.0           89  
36238                Aaptar       5.0          148

### Basic information of DataFrame
- We can see total rows 36239
- Total School_name rows equals to 36237
- some of them don't have ward number location 
- we need to fill Nan with some values
- For Location  we will fill NaN with "VDC/Municipality_name" 
- For Ward No. We will Fill NaN with 0

In [12]:
school_data_df_copy.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 36239 entries, 0 to 36238
Data columns (total 6 columns):
School_name              36237 non-null object
District_name            36237 non-null object
Location                 36003 non-null object
VDC/Municipality_name    36237 non-null object
Ward No.                 36129 non-null float64
Unnamed: 125             36239 non-null object
dtypes: float64(1), object(5)
memory usage: 1.7+ MB


In [13]:
## droping first two rows
school_data_df1 = school_data_df_copy.drop(school_data_df_copy.index[[0,1]])

school_data_df1.head()

School_name District_name               Location  \
2  Adarsha Ni Ma Vi        ACHHAM                 Kuskot   
3     Adarsha Pra V        ACHHAM              Badchauki   
4   Alaknanda Ma Vi        ACHHAM              Tinilsain   
5      Ambika Pra V        ACHHAM              Pulletala   
6       Ambika Ma V        ACHHAM  Nanakhal Salyan Danda   

  VDC/Municipality_name  Ward No. Unnamed: 125  
2                Kuskot       2.0          187  
3             Timilsain       4.0           82  
4             Timilsain       5.0          460  
5             Pulletola       8.0          143  
6           Kalikasthan       3.0          351

#### Basic info of New DataFrame

In [14]:
school_data_df1.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 36237 entries, 2 to 36238
Data columns (total 6 columns):
School_name              36237 non-null object
District_name            36237 non-null object
Location                 36003 non-null object
VDC/Municipality_name    36237 non-null object
Ward No.                 36129 non-null float64
Unnamed: 125             36237 non-null object
dtypes: float64(1), object(5)
memory usage: 1.9+ MB


In [15]:
##filling Nan with values
school_data_df1['Location'] = school_data_df1['Location'].fillna(school_data_df1['VDC/Municipality_name'])
school_data_df1['Ward No.'] = school_data_df1['Ward No.'].fillna(0)

In [16]:
school_data_df1.head()

School_name District_name               Location  \
2  Adarsha Ni Ma Vi        ACHHAM                 Kuskot   
3     Adarsha Pra V        ACHHAM              Badchauki   
4   Alaknanda Ma Vi        ACHHAM              Tinilsain   
5      Ambika Pra V        ACHHAM              Pulletala   
6       Ambika Ma V        ACHHAM  Nanakhal Salyan Danda   

  VDC/Municipality_name  Ward No. Unnamed: 125  
2                Kuskot       2.0          187  
3             Timilsain       4.0           82  
4             Timilsain       5.0          460  
5             Pulletola       8.0          143  
6           Kalikasthan       3.0          351

In [17]:
school_data_df1.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 36237 entries, 2 to 36238
Data columns (total 6 columns):
School_name              36237 non-null object
District_name            36237 non-null object
Location                 36237 non-null object
VDC/Municipality_name    36237 non-null object
Ward No.                 36237 non-null float64
Unnamed: 125             36237 non-null object
dtypes: float64(1), object(5)
memory usage: 1.9+ MB


##### converting ward No and Unnamed:125 data type to int

In [18]:
school_data_df1[['Ward No.','Unnamed: 125']] = school_data_df1[['Ward No.','Unnamed: 125']].astype(int)

In [19]:
school_data_df2 = school_data_df1.rename(columns={"Unnamed: 125":"Total Sutdents: 2070"})

In [20]:
school_data_df2.head()

School_name District_name               Location  \
2  Adarsha Ni Ma Vi        ACHHAM                 Kuskot   
3     Adarsha Pra V        ACHHAM              Badchauki   
4   Alaknanda Ma Vi        ACHHAM              Tinilsain   
5      Ambika Pra V        ACHHAM              Pulletala   
6       Ambika Ma V        ACHHAM  Nanakhal Salyan Danda   

  VDC/Municipality_name  Ward No.  Total Sutdents: 2070  
2                Kuskot         2                   187  
3             Timilsain         4                    82  
4             Timilsain         5                   460  
5             Pulletola         8                   143  
6           Kalikasthan         3                   351

In [24]:
school_data_df2.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 36237 entries, 2 to 36238
Data columns (total 6 columns):
School_name              36237 non-null object
District_name            36237 non-null object
Location                 36237 non-null object
VDC/Municipality_name    36237 non-null object
Ward No.                 36237 non-null int64
Total Sutdents: 2070     36237 non-null int64
dtypes: int64(2), object(4)
memory usage: 1.9+ MB


# Savinng Data Frame  to CSV

In [25]:
location = "data/clean-data-all-schools-list.csv"
school_data_df2.to_csv(location,index=False)

In [26]:
pd.read_csv(location)

School_name District_name  \
0                            Adarsha Ni Ma Vi        ACHHAM   
1                               Adarsha Pra V        ACHHAM   
2                             Alaknanda Ma Vi        ACHHAM   
3                                Ambika Pra V        ACHHAM   
4                                 Ambika Ma V        ACHHAM   
5                        Annapurna Uchha Ma V        ACHHAM   
6                             Arunodaya Pra V        ACHHAM   
7                                  Asrya Ma V        ACHHAM   
8                            Asta Siddhi Ma V        ACHHAM   
9                           Badhimalika Pra V        ACHHAM   
10                          Bageswori Ni Ma V        ACHHAM   
11                           Bageshwori Pra V        ACHHAM   
12                           Bageshwori Pra V        ACHHAM   
13                         Bageswori Ni Ma Vi        ACHHAM   
14                           Bageshwori Pr Vi        ACHHAM   
15                               Baguda Pra V        ACHHAM   
16          Baidyanath Bed Bidyashram Ni Ma V        ACHHAM   
17                       Baijanath Uchha Ma V        ACHHAM   
18                      Baikuntheshwari Pra V        ACHHAM   
19                           Bal Bikash Pra V        ACHHAM   
20                           Bal Bikash Pra V        ACHHAM   
21                           Bal Bikash Pra V        ACHHAM   
22                           Bal Bikash Pra V        ACHHAM   
23                            Bal Jyoti Pra V        ACHHAM   
24                           Bal Kalyan Pra V        ACHHAM   
25                           Bal Kalyan Pra V        ACHHAM   
26                           Bal Kalyan Pra V        ACHHAM   
27                            Balmandir Pra V        ACHHAM   
28                           Balbatika Pra Vi        ACHHAM   
29                              Balkalyan P S        ACHHAM   
...                                       ...           ...   
36207             Singhakali Ls Pakanactumchi      UDAYAPUR   
36208                    Sisaghari Ebs Katari      UDAYAPUR   
36209  Sivasakti Ps Bhimsentol Saune Choutara      UDAYAPUR   
36210         Siwalaya Ps Dhauri Tappu Rampur      UDAYAPUR   
36211                            Sudarsan Ebs      UDAYAPUR   
36212                     Sudarsan Ebs Hadiya      UDAYAPUR   
36213               Sukadeb Memoriyal Acedamy      UDAYAPUR   
36214                 Sungava Pra V Tinchocke      UDAYAPUR   
36215                Sunkashi Ps Katarhanjang      UDAYAPUR   
36216                   Sunkoshi Pra V Kunahi      UDAYAPUR   
36217                   Suryodaya Ps Khiratar      UDAYAPUR   
36218                    Suvatara Pra V Papre      UDAYAPUR   
36219            Tapeswori Ni Ma V Galphariya      UDAYAPUR   
36220                      Thana Pokhari Ma V      UDAYAPUR   
36221     Triveni Educational Academy Tirveni      UDAYAPUR   
36222                      Triveni Uchha Ma V      UDAYAPUR   
36223               Triyuga Public Ebs Hadiya      UDAYAPUR   
36224                      Triyuga Uchha Ma V      UDAYAPUR   
36225                            U Ma V Barre      UDAYAPUR   
36226                Uchha Ma V Bhalaya Danda      UDAYAPUR   
36227            Uchha Ma V Chilaunebhanjyang      UDAYAPUR   
36228                      Uchha Ma V Hadebas      UDAYAPUR   
36229   Higher Secondary School Jhilkepokhari      UDAYAPUR   
36230                     Uchha Ma V Laxmipur      UDAYAPUR   
36231                     Uchha Ma V Murkuchi      UDAYAPUR   
36232                        U Ma V Sundarpur      UDAYAPUR   
36233                   Udaya Pra V Salthumki      UDAYAPUR   
36234            Udayapur Little Heaven E B S      UDAYAPUR   
36235                   Unik Ebs Shishu Sahan      UDAYAPUR   
36236                 Yuba Ni Ma V Dhukurthum      UDAYAPUR   

                    Location VDC/Municipality_name  Ward No.  \
0                     Kuskot                Kuskot         2   